add repository update frequency documentation (#6204)
sourcegraph/sourcegraph: doc/admin/repo/update_frequency.md

# Repository update frequency

> Sourcegraph polls code hosts to keep repository contents up to date by default, effectively periodically running `git pull`. You can also configure Sourcegraph to use [repository webhooks](webhooks.md) instead, but this is generally not needed.

Reviewer: ```suggestion
By default, Sourcegraph polls code hosts to keep repository contents up to date, effectively running `git pull` periodically. You can also configure Sourcegraph to use [repository webhooks](webhooks.md), but this is usually not necessary.
```

> The frequency at which Sourcegraph polls the code host for updates is determined by a smart hueristic based on past commit frequency in the repository. For example, if a repositories last commit was 8 hours ago, then the next update will be scheduled 4 hours from now. If there are still no new commits, then the next update will be scheduled 6 hours from then.

Reviewer 2: heuristic

> The frequency at which Sourcegraph polls the code host for updates is determined by a smart hueristic based on past commit frequency in the repository. For example, if a repositories last commit was 8 hours ago, then the next update will be scheduled 4 hours from now. If there are still no new commits, then the next update will be scheduled 6 hours from then.

Reviewer: ```suggestion
The frequency at which Sourcegraph polls the code host for updates is determined by a smart heuristic based on past commit frequency in the repository. For example, if a repository's last commit was 8 hours ago, then the next sync will be scheduled 4 hours from now. If after 4 hours, there are still no new commits, then the next update will be scheduled 6 hours from then.
```

Reviewer: Can we say what sort of backoff it is? Linear, exponential, or somewhere in between?

Reviewer 3: I would use the word `fetch` instead of `sync` or `update`

> The frequency at which Sourcegraph polls the code host for updates is determined by a smart heuristic based on past commit frequency in the repository. For example, if a repository's last commit was 8 hours ago, then the next sync will be scheduled 4 hours from now. If after 4 hours, there are still no new commits, then the next update will be scheduled 6 hours from then.

Author: ```suggestion
The frequency at which Sourcegraph polls the code host for updates is determined by a smart heuristic based on past commit frequency in the repository. For example, if a repository's last commit was 8 hours ago, then the next sync will be scheduled 4 hours from now. If after 4 hours, there are still no new commits, then the next sync will be scheduled 6 hours from then.
```

Repositories will never be updated more frequently than 45 seconds, and no less frequently than every 8 hours.

> After Sourcegraph has polled the code host for Git updates, the search index will automatically update a short while after (usually taking just a few minutes per repository).

Reviewer: ```suggestion
After Sourcegraph has updated a repository's Git data, the global search index will automatically update a short while after (usually a few minutes).
```

## Limiting repository updates

If you wish to control how frequently repositories are discovered or how frequently Sourcegraph polls your code host for updates, tuning parameters are available in the site configuration:

> - [repoListUpdateInterval](../config/site_config.md#repoListUpdateInterval) controls how frequently we check the code host _for new repositories_ in minutes. Defaults to 1 minute.

Reviewer: ```suggestion
- [repoListUpdateInterval](../config/site_config.md#repoListUpdateInterval) controls how frequently we check the code host _for new repositories_ in minutes.
```

^ I think it's better to let the user click through to learn what the default is.

- [gitMaxConcurrentClones](../config/site_config.md#gitMaxConcurrentClones) controls the maximum number of _concurrent_ cloning / pulling operations that Sourcegraph will perform.

> You may also choose to disable automatic Git updates entirely and instead [configure repository webhooks](webhooks.md).

Reviewer 3: ```suggestion
You may also choose to [disable automatic Git updates entirely](https://docs.sourcegraph.com/admin/config/site_config#disableAutoGitUpdates) and instead [configure repository webhooks](webhooks.md).
```

Author: I intentionally didn't link to this, the webhook documentation already describes this in more depth.
